apiserver/apiservermetrics.go

```diff
@@ -4,81 +4,88 @@
 package apiserver
 
 import (
-	"time"
-
+	"github.com/juju/juju/apiserver/observer/metricobserver"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
 const (
-	apiserverMetricsNamespace = "juju_apiserver"
+	apiserverMetricsNamespace = "juju"
 )
 
-// ServerMetricsSource implementations provide apiserver metrics.
-type ServerMetricsSource interface {
-	TotalConnections() int64
-	ConnectionCount() int64
-	ConcurrentLoginAttempts() int64
-	ConnectionPauseTime() time.Duration
-}
+const (
+	SubsystemNamespace           = "apiserver"
+	DeprecatedSubsystemNamespace = "api"
+)
 
 // Collector is a prometheus.Collector that collects metrics based
 // on apiserver status.
 type Collector struct {
-	src ServerMetricsSource
-
-	connectionCounter        prometheus.Counter
-	connectionCountGauge     prometheus.Gauge
-	connectionPauseTimeGauge prometheus.Gauge
-	concurrentLoginsGauge    prometheus.Gauge
+	TotalConnections   prometheus.Counter
+	APIConnections     prometheus.Gauge
+	LogsinkConnections prometheus.Gauge
+	LoginAttempts      prometheus.Gauge
+	APIRequestsTotal   *prometheus.CounterVec
+	APIRequestDuration *prometheus.SummaryVec
 }
 
 // NewMetricsCollector returns a new Collector.
-func NewMetricsCollector(src ServerMetricsSource) *Collector {
+func NewMetricsCollector(subsystem string) *Collector {
 	return &Collector{
-		src: src,
-		connectionCounter: prometheus.NewCounter(prometheus.CounterOpts{
+		TotalConnections: prometheus.NewCounter(prometheus.CounterOpts{
 			Namespace: apiserverMetricsNamespace,
+			Subsystem: subsystem,
 			Name:      "connections_total",
 			Help:      "Total number of apiserver connections ever made",
 		}),
-		connectionCountGauge: prometheus.NewGauge(prometheus.GaugeOpts{
+		APIConnections: prometheus.NewGauge(prometheus.GaugeOpts{
 			Namespace: apiserverMetricsNamespace,
+			Subsystem: subsystem,
 			Name:      "connection_count",
 			Help:      "Current number of active apiserver connections",
 		}),
-		connectionPauseTimeGauge: prometheus.NewGauge(prometheus.GaugeOpts{
+		LogsinkConnections: prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
I thought you were considering having a label for the connection type? Are we not doing that now?

Reply:
I'll put that back in, sorry it seems I got my wires crossed :(